app/file_test.go

```diff
@@ -299,3 +300,32 @@ func TestCopyFileInfos(t *testing.T) {
 	assert.NotEqual(t, info1.Id, info2.Id, "should not be equal")
 	assert.Equal(t, info2.PostId, "", "should be empty string")
 }
+
+func TestGenerateThumbnailImage(t *testing.T) {
+	t.Run("test generating thumbnail image", func(t *testing.T) {
+		// given
+		th := Setup(t)
+		defer th.TearDown()
+		tempDir := os.TempDir()
+		img := createDummyImage()
+
+		// when
+		th.App.generateThumbnailImage(img, tempDir)
+
+		// then
+		absolutePath := fmt.Sprintf("%v/___TestApp_generateThumbnailImage_test_generating_thumbnail_image_in_github_com_mattermost_mattermost_server_v5_app", tempDir)
```

Comment:
I think we should pass the actual output filepath to `generateThumbnailImage()` and test on that, e.g.:

```go
		thumbnailPath := tempDir + "/thumb.jpg"
		th.App.generateThumbnailImage(img, thumbnailPath)

		// then
		if outputImage, err := os.Stat(thumbnailPath); err == nil {
			assert.Equal(t, int64(69322349), outputImage.Size())
			defer os.Remove(thumbnailPath)
		}
```

Reply:
i've change the test a little bit and passing the `thumbnailPath` to the `generateThumbnailImage` doesnt' work, because the name of the generated thumbnail is something like `___1TestGenerateThumbnailImage_test_generating_thumbnail_image_in_github_com_mattermost_mattermost_server_v5_app`. The output name depends on the test.

Reply:
Apologies. The data is written by the store to some pre-configured path (by default under the `data` dir).
So we need to get the path of that and then just pass the filename to `generateThumbnailImage()`. 

```go
		// given
		th := Setup(t)
		defer th.TearDown()
		img := createDummyImage()
		dataPath, _ := fileutils.FindDir("data")
		thumbailName := "thumb.jpg"
		thumbnailPath := filepath.Join(dataPath, thumbailName)

		// when
		th.App.generateThumbnailImage(img, thumbailName)
		defer os.Remove(thumbnailPath)

		// then
		outputImage, err := os.Stat(thumbnailPath)
		assert.NoError(t, err)
		assert.Equal(t, int64(957), outputImage.Size())
```

Reply:
thanks a lot, imho it looks good now